sources/models/src/modeled_types/mod.rs

```diff
@@ -158,7 +158,7 @@ macro_rules! string_impls_for {
             type Error = $crate::modeled_types::error::Error;
 
             fn try_from(input: String) -> Result<Self, Self::Error> {
-                Self::try_from(input.as_ref())
+                input.parse::<Self>()
```

Comment:
We should update the module-level docstring, which provides some information about `TryFrom<&str>` and how this macro behaves.

Reply:
How does this sound? (Just changed the `TryFrom<&str>` to `FromStr` -- the rest looks accurate?)

```rust
// The pattern in this module is to make a struct and implement FromStr with code that does
// necessary checks and returns the struct.  Other traits that treat the struct like a string can
// be implemented for you with the string_impls_for macro.
```